soroban-env-host/src/host/error.rs

```diff
@@ -254,7 +254,13 @@ impl Host {
     /// enriches the returned [Error] with [DebugInfo] in the form of a
     /// [Backtrace] and snapshot of the [Events] buffer.
     pub(crate) fn error(&self, error: Error, msg: &str, args: &[Val]) -> HostError {
-        let mut he = HostError::from(error);
+        self.secondary_error(HostError::from(error), msg, args)
+    }
```

Comment:
This is just an implementation detail, right? Maybe it's worth to add a comment stating that `secondary_error` here doesn't actually mean it's 'secondary', i.e. the input error will have its backtrace preserved.

Reply:
Yes. I have pushed explanatory comments as requested.